gramps/gen/lib/primaryobj.py

```diff
@@ -70,87 +70,62 @@ def __init__(self, source=None):
         of the source object.
 
         :param source: Object used to initialize the new object
-        :type source: PrimaryObject
         """
         TableObject.__init__(self, source)
         PrivacyBase.__init__(self, source)
         TagBase.__init__(self)
-        if source:
-            self.gramps_id = source.gramps_id
-        else:
-            self.gramps_id = None
+        self.gramps_id = source.get_gramps_id() if source else ""
 
-    @abstractmethod
-    def serialize(self):
-        """
-        Convert the object to a serialized tuple of data.
-        """
-
-    @abstractmethod
-    def unserialize(self, data):
-        """
-        Convert a serialized tuple of data to an object.
-        """
-
-    def set_gramps_id(self, gramps_id):
+    def set_gramps_id(self, gramps_id: str) -> None:
         """
         Set the Gramps ID for the primary object.
 
         :param gramps_id: Gramps ID
-        :type gramps_id: str
         """
         self.gramps_id = gramps_id
 
-    def get_gramps_id(self):
+    def get_gramps_id(self) -> str:
         """
         Return the Gramps ID for the primary object.
 
         :returns: Gramps ID associated with the object
-        :rtype: str
         """
         return self.gramps_id
 
-    def has_handle_reference(self, classname, handle):
+    def has_handle_reference(self, classname: str, handle: str) -> bool:
         """
         Return True if the object has reference to a given handle of given
         primary object type.
 
         :param classname: The name of the primary object class.
-        :type classname: str
         :param handle: The handle to be checked.
-        :type handle: str
 
         :returns:
           Returns whether the object has reference to this handle of
           this object type.
-
-        :rtype: bool
         """
         return False
 
-    def remove_handle_references(self, classname, handle_list):
+    def remove_handle_references(self, classname: str, handle_list: list[str]) -> None:
```

Comment:
We probably want to widen the argument type and avoid `list`. Possibly `Iterable` because I don't think order matters for removal?

Reply:
@dsblank @Nick-Hall Mind if I remove 'types' from arguments names? E.g. `handles` instead of `handle_list`? Just so readers won't get confused.